restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -226,7 +226,8 @@ private boolean isEmptyArray(Object v) {
                     if (pv.equals("$")) continue;
 
                     // pv always starts with a square brace because of the above conversion
-                    String innerPath = pv.substring(1, pv.length() - 1);
+                    String innerPath =
+                        DocsApiUtils.convertUnicodeCodePoints(pv.substring(1, pv.length() - 1));
```

Comment:
I'm not sure I understand why we need to unescape `pv` in this case :thinking: Could you clarify, please?

I assume we always take any JSON name in a doc body "as is" and only apply escaping to names from query parameters (e.g. field paths). Is that a valid assumption?

Reply:
I was hoping for that to be the case, but because of what I named above https://github.com/stargate/stargate/pull/1105#discussion_r677464700 there is actually a need to handle escaping while writing a document as well.

For example, a document such as `{ "a.b": "c" }` will be converted by the JsonSurfer library to `$.a.b` (which isn't quite right!!) - and then our code turns that into `$['a']['b']` - essentially this will write `p0='a', p1='b', text_value='c'` to cassandra, which isn't quite what we want here (we would want `p0='a.b', text_value='c'`)

So, by escaping like `{ "a\u002eb": "c" }`, the JsonSurfer library makes the path `$.a\u002eb`, and we convert it to `$['a\u002eb']` - which would cause `p0='a\u002eb', text_value='c'` to be written to the db; close but not quite right.

So the last thing that has to be done is to take each of pN and convert unicode escape sequences to actual literal characters, and then it's all correct in the database.